plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcTableHandle.java

```diff
@@ -132,7 +157,7 @@ public OptionalLong getLimit()
         return columns;
     }
 
-    private JdbcNamedRelationHandle getNamedRelation()
+    public JdbcNamedRelationHandle getNamedRelation()
```

Comment:
I don't like this method. It is unsafe and should not be used as public. Failing at runtime is a bit too late. Why not to return `Optional` here?

Reply:
you're expected to call this only when it's a named relation (hence the name), so it's as safe as `getXxx().orElseThrow()`

Reply:
I don't find this name `getXxxYyy` suggests that method is a plain getter, unless you are familiar with this code.
I bet people will forget about calling `isNamedRelation()` before calling this method.
And if I am required to call `isNamedRelation()` before calling `getNamedRelation()` then it is much easier to return `Optional` and skip `isNamedRelation()`

Reply:
I am not sure if it would turn out to be a problem. We have many places in code when you cannot call on of the methods defined in interface. E.g. `getLong`, `getDouble` ... in types.

But I see your point @kokosing . I think the aproach with `Optional` is nice, and should not be very painful at call spots.

Reply:
i propose to call it `getRequiredNamedRelation` to avoid confusion

Reply:
see also that we have `isNamedRelation` already and we use it in a few places (for a good reason)

Reply:
i take it back. I don't propose to add "Required". 
There is fundamental problem that Table Handle represent a table and also synthetic relation.
This is a by-product of the fact how we structured #18 work and i do not see a better alternative, 
but we have to leave with the consequences. Which is that we ensure & extract the named relation
from a table handle in sooo many places.

Actually, in those cases we want to have a different accessor, that would validate the relation is not synthetic.
I will change the code appropriately.

Reply:
I don't like that name very much. I was trying to think of something I like yesterday ... no success. I think `Optional` may capture the intention better than the name proposed, but I will not fight for that.

Reply:
I feel strongly that Optional here is going to be code obfuscator. I pushed 2 commits where i updated deprecated call sites, so that you can get a feel how this would look like if i switched to `optional` + `orElseThrow(() -> new IllegalStateException("some useful exception message quoting " + tableHandle));`.
It's 40% done, as there are 3 more deprecated methods to be udpated (each with roughtly similar number of call sites).